proposals/0000-astro-url-helper.md

```diff
@@ -0,0 +1,118 @@
+- Start Date: 2022-07-18
+- Reference Issues: N/A
+- Implementation PR: https://github.com/withastro/astro/pull/3959
+
+# Summary
+
+Create a general-purpose `Astro.url` helper. Deprecate `Astro.canonicalURL` to use `Astro.url` instead.
```

Comment:
This is an overall comment on the RFC, I think the framing as this is a replacement/renaming makes it seem like they are close to the same feature, but they aren't really. 

I think it's more accurate to say that you're adding a new feature, `Astro.url`, is has more general usage in more scenarios and removing Astro.canonicalURL, that has 1 specific use-case.

It's now slightly harder to produce a canonicalURL but it's easier to get the current pathname, etc, things you need more often. I think the examples illustrating this are good.

Reply:
Yup, let me see if I can hit this point better. I like the way you've phrased it in this comment.

Reply:
added this to the Drawback section and reframed the Motivation section to better frame the RFC as you did.